src/governance.h

```diff
@@ -147,11 +163,19 @@ class CGovernanceManager
     void UpdatedBlockTip(const CBlockIndex *pindex);
     int64_t GetLastDiffTime() {return nTimeLastDiff;}
     void UpdateLastDiffTime(int64_t nTimeIn) {nTimeLastDiff=nTimeIn;}
+
+    int GetCachedBlockHeight() { return nCachedBlockHeight; }
```

Comment:
If it's done in such a way that I can grab the current block height without locking cs_main that would be perfect.  Is there an example of how that can be done in the code somewhere so I can see how to use that mechanism ?